### PR TITLE
Tests for bid adjustment factors

### PR DESCRIPTION
**Pull request by Net-burst**
Tests for https://github.com/prebid/prebid-server-java/pull/1865

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+        and: "Default bid response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Final bid price should be adjusted"
+        assert response?.seatbid[0]?.bid[0]?.price == bidResponse.seatbid[0].bid[0].price * bidAdjustmentFactor
```
Comment (hupaloo) [inline src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy:34]: If seatbid is null, so then seatbid[0] will throw a NPE. Groovy null-safe operator can't help when you access an element with an array index operator. To guarantee a null-safety '?.' operator should be used as seatbid?.first().
Same for other places

File: src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        where:
+        bidAdjustmentFactor << [0.9, 1.1]
+    }
+
+    def "PBS should prefer bid price adjustment based on media type when request has per-media-type bid adjustment factors"() {
+        given: "Default bid request with bid adjustment"
+        def bidAdjustment = PBSUtils.getFractionalRandomNumber() as BigDecimal
```
Comment (hupaloo) [inline src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy:42]: you may use a Groovy shortcut notation to call getFractionalRandomNumber()

File: src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        then: "Final bid price should be adjusted"
+        assert response?.seatbid[0]?.bid[0]?.price == bidResponse.seatbid[0].bid[0].price * mediaTypeBidAdjustment
+
+        where:
+        bidAdjustmentFactor << [0.9, 1.1]
+    }
+
+    def "PBS should adjust bid price for bidder only when request contains bid adjustment for corresponding bidder"() {
+        given: "Default bid request with bid adjustment"
+        def bidAdjustment = PBSUtils.randomFractionalNumber as BigDecimal
```
Comment (hupaloo) [inline src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy:67]: method does not exist, should be fractionalRandomNumber

File: src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy (hunk 1 above)
Comment (Net-burst) [reply src/test/groovy/org/prebid/server/functional/tests/BidAdjustmentSpec.groovy:34]: Derp. Good catch.

**Review by hupaloo — APPROVED** [review]

**Pull request merged**